File: src/main/java/com/github/ccguyka/showupdates/producer/ArtifactSource.java
package com.github.ccguyka.showupdates.producer;

import java.util.ArrayList;
import java.util.List;

import org.apache.maven.artifact.Artifact;
import org.apache.maven.artifact.factory.ArtifactFactory;
import org.apache.maven.model.Dependency;

public class ArtifactSource {

    private final ArtifactFactory artifactFactory;

    public ArtifactSource(final ArtifactFactory artifactFactory) {
        this.artifactFactory = artifactFactory;
    }

    public List<Artifact> getArtifacts(final List<Dependency> dependencies) {

        final List<Artifact> artifacts = new ArrayList<>();
        dependencies.stream().forEach(dependency -> artifacts.add(createArtifactFrom(dependency)));
        return artifacts;
    }

    private Artifact createArtifactFrom(final Dependency dependency) {
        return artifactFactory.createArtifact(dependency.getGroupId(), dependency.getArtifactId(),
                dependency.getVersion(), dependency.getScope(), dependency.getType());
    }
}

File: src/main/kotlin/com/github/ccguyka/showupdates/producer/ArtifactSource.kt
package com.github.ccguyka.showupdates.producer

import org.apache.maven.artifact.Artifact
import org.apache.maven.artifact.factory.ArtifactFactory
import org.apache.maven.model.Dependency
import java.util.*

class ArtifactSource(private val artifactFactory: ArtifactFactory) {

    fun getArtifacts(dependencies: List<Dependency>): List<Artifact> {
        val artifacts: MutableList<Artifact> = ArrayList()
        dependencies.stream().forEach { dependency: Dependency -> artifacts.add(createArtifactFrom(dependency)) }
        return artifacts
    }

    private fun createArtifactFrom(dependency: Dependency): Artifact {
        return artifactFactory.createArtifact(dependency.groupId, dependency.artifactId,
                dependency.version, dependency.scope, dependency.type)
    }
}
